### PR TITLE
Buffs The CMO's Advanced Hypospray

### PR DESCRIPTION
**Pull request by Fordoxia**
## What Does This PR Do
Allows the CMO's Advanced Hypospray to work through hardsuits.

Also increases the reagent capacity to 100u to hammer home the utility over the basic hypo outside of hardsuit stuff (the amount of omnizine the spray starts with has not been changed).

Also does some documentation, and renames a really badly named variable.

The CMO hypo and emagged hyposprays cannot penetrate nukeops suits, because that would be very silly. Their own hyposprays can, though. ::)

## Why It's Good For The Game
The CMO's hypo is actually kind of lame. We have a modsuit hypo that can inject through hardsuits, and regular hypos can also do it if they are emagged (the advanced hypo, however, cannot be emagged).

In the hands of a non-antag, the CMO hypo is pretty much identical to the vendor hypo, and antags have access to better options for covertly filling you with drugs. It's lame.

This makes it an obviously BETTER item, CMOs will actually have a very strong reason to want to hold onto it now because it's a proper, compelling upgrade to the basic hypo.

## Testing
It compiled.
I injected myself through a hardsuit.
I could not inject through a nukie suit without a nukie hypo.

## Changelog
:cl:
tweak: CMO hypo can now pierce hardsuits.
/:cl:

### CODE REVIEW

**Comment by Adrer** [comment]
I forsee an issue regarding nukies.

**Comment by Butlerblock** [comment]
> I forsee an issue regarding nukies.

Could snowflake it so it doesn't inject through certain hardsuits

**Comment by Fordoxia** [comment]
> I forsee an issue regarding nukies.

The issue is gone.

File: code/modules/reagents/reagent_containers/hypospray.dm [hunk 1]
```diff
@@ -15,8 +14,13 @@
...
 	container_type = OPENCONTAINER
 	slot_flags = SLOT_FLAG_BELT
-	var/ignore_flags = FALSE
+	/// If TRUE, the hypospray can inject through most hardsuits/modsuits.
+	var/can_pierce_hardsuits = FALSE
+	/// If TRUE, the hypospray isn't blocked by suits with TRAIT_HYPOSPRAY_IMMUNE.
+	var/ignore_hypospray_immunity = FALSE
+	/// If TRUE, the hypospray will reject any chemicals not on the safe_chem_list.
 	var/safety_hypo = FALSE
+	/// List of OSHA-approved medicines.
```
Comment (DGamerL) [inline code/modules/reagents/reagent_containers/hypospray.dm:23]:
```suggestion
	/// List of SOSHA-approved medicines.
```
Alternatively,
```suggestion
	// List of SOSHA-approved medicines.
	/// List of reagents that are allowed to go into a hypospray with active safeties
```

**Review by DGamerL — CHANGES_REQUESTED** [review]
Also add your trait to `traits.dm`

**Comment by Fordoxia** [comment]
> Also add your trait to `traits.dm`

Done.

**Review by DGamerL — APPROVED** [review]

**Review by lewcc — APPROVED** [review]

**Pull request merged**